### ISSUE TITLE
Closures with variables defined later give wrong warnings

### ISSUE THREAD

**Issue by fredrikekre** [julia-vscode/StaticLint.jl#313]
```
function f()
    function g()
        println("hello, $(who)")    # <- missing reference who
    end
    who = "world"                   # <- assigned by unused variable
    g()
end
```
